### PR TITLE
Automatically expand a section even after page load

### LINKED ISSUE #52774: docs: dead link to std::vec::Vec::extend()
[````extend_from_slice````](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice) links to [````extend````](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend) however it seems there is no such method in the docs.

### PR DESCRIPTION
**Pull request by kzys**
Fixes #52774

### CODE REVIEW

File: src/librustdoc/html/static/main.js [hunk 1]
```diff
@@ -223,8 +223,38 @@
...
     }
+
+    function expandSection() {
+        var hash = getPageId();
+        if (hash === null) {
+            return;
+        }
+
+        var elem = document.getElementById(hash);
+        if (elem && isHidden(elem.offsetParent)) {
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/main.js]: `offsetParent` can be `null`, please check it before calling `isHidden`.

**Comment by GuillaumeGomez** [comment]
Two nits to fix and it's good to go.

File: src/librustdoc/html/static/main.js [hunk 2]
```diff
@@ -223,8 +223,38 @@
...
+            var h3 = elem.parentNode.previousSibling;
+
+            if (h3.tagName !== 'H3') {
```
[the remark this answers is not in the document]
Comment (kzys) [reply src/librustdoc/html/static/main.js]: Thanks. Done at 1f441a0

File: src/librustdoc/html/static/main.js (hunk 1 above)
Comment (kzys) [reply src/librustdoc/html/static/main.js]: Ditto

**Comment by ehuss** [comment]
Would it be possible to fix the issues listed in #48726?  It looks like this is close, but doesn't seem to work on those examples.

**Comment by GuillaumeGomez** [comment]
@ehuss This PR should fix it.

**Comment by GuillaumeGomez** [comment]
@kzys Thanks a lot for your work!

@bors: r+ rollup

**Comment by ehuss** [comment]
I tried this patch, it doesn't seem to work.  
* In the example of BufWriter, the `flush` link doesn't work because it is nested within a trait implementation, so it actually needs to check recursively up the parents (document.getElementById("method.flush").offsetParent.offsetParent.offsetParent).
* In the example of hiding all methods, and then trying to click on the sidebar, the immediate offsetParent is null.
* When using cross-page links, AFAIK these won't trigger.  The same logic needs to be applied somewhere like `autoCollapse` to prevent them from being collapsed in the first place (or immediately reopen them).

**Comment by GuillaumeGomez** [comment]
@ehuss: Thanks for your check!
@kzys: Can you fix these issues too please?

@bors: r-

**Comment by kzys** [comment]
@ehuss The last change (2c61f3c) should address the issues you've mentioned.

For cross-page links, I couldn't reproduce the issue even before my fix. How did you check the behavior?

**Comment by ehuss** [comment]
I tried a variety of things, and everything seems to work now!  TYVM!!

**Comment by GuillaumeGomez** [comment]
Thanks a lot again, let's get this in!

@bors: r+ rollup

**Pull request merged**